cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -681,6 +683,16 @@ __device__ inline bool is_literal_run(int const run_header) { return (run_header
  */
 __device__ inline bool is_repeated_run(int const run_header) { return !is_literal_run(run_header); }
 
+/**
+ * @brief Launches kernel for counting page headers from column chunk data buffers
+ *
+ * @param[in] chunks Device span of column chunks
+ * @param[out] error_code Error code for kernel failures
+ * @param[in] stream CUDA stream to use
+ */
+void count_page_headers(cudf::detail::hostdevice_span<ColumnChunkDesc> chunks,
```

Comment:
Use a separate dedicated function to count page headers instead of calling `decode_page_headers` with a certain set of args.